README.md

```diff
@@ -0,0 +1,178 @@
+      id-token: write
+      contents: read
+    uses: slsa-framework/slsa-github-generator/.github/workflows/provenance.yml@main
+    with:
+      subjects: "${{ needs.build.outputs.digest }}"
+```
+
+### Workflow Inputs
+
+The builder workflow
+[.github/workflows/provenance.yml](.github/workflows/provenance.yml) accepts
+the following inputs:
+
+| Name       | Required | Description                                                                                                    |
+| ---------- | -------- | -------------------------------------------------------------------------------------------------------------- |
+| `subjects` | yes      | Artifacts for which to generate provenance, formatted the same as the output of sha256sum (SHA256 NAME\n[...]) |
+
+### Provenance Example
+
+The following is an example of the generated proveanance. Provenance is
+generated as an [in-toto](https://in-toto.io/) statement with a SLSA predecate.
+
+```json
+{
+  "_type": "https://in-toto.io/Statement/v0.1",
+  "predicateType": "https://slsa.dev/provenance/v0.2",
+  "subject": [
+    {
+      "name": "binary-linux-amd64",
+      "digest": {
+        "sha256": "2e0390eb024a52963db7b95e84a9c2b12c004054a7bad9a97ec0c7c89d4681d2"
+      }
+    },
+  ],
+  "predicate": {
+    "builder": {
+      "id": "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/provenance.yml@refs/heads/main"
+    },
+    "buildType": "https://github.com/slsa-framework/slsa-github-generator@v1",
+    "invocation": {
+      "configSource": {
+        "uri": "git+https://github.com/ianlewis/actions-test@refs/heads/main.git",
+        "digest": {
+          "sha1": "3b5dc7cf5c0fd71c5a74c6b16cae78d49e03857c"
+        },
+        "entryPoint": "SLSA provenance"
+      },
+      "parameters": {},
+      "environment": {
+        "github_actor": "ianlewis",
+        "github_base_ref": "",
+        "github_event_name": "workflow_dispatch",
+        "github_event_payload": ...,
+        "github_head_ref": "",
+        "github_ref": "refs/heads/main",
+        "github_ref_type": "branch",
+        "github_run_attempt": "1",
+        "github_run_id": "2093917134",
+        "github_run_number": "19",
+        "github_sha1": "3b5dc7cf5c0fd71c5a74c6b16cae78d49e03857c"
+      }
+    },
+    "metadata": {
+      "completeness": {
+        "parameters": true,
+        "environment": false,
+        "materials": false
+      },
+      "reproducible": false
+    },
+    "materials": [
+      {
+        "uri": "git+https://github.com/ianlewis/actions-test@refs/heads/main.git",
+        "digest": {
+          "sha1": "3b5dc7cf5c0fd71c5a74c6b16cae78d49e03857c"
+        }
+      }
+    ]
+  }
+}
+```
+
+## Verification of Provenance
+
+To verify the provenance, use the
+[github.com/slsa-framework/slsa-verifier](https://github.com/slsa-framework/slsa-verifier)
+project.
+
+### Inputs
+
+```shell
+$ git clone git@github.com:slsa-framework/slsa-verifier.git
+$ go run . --help
```

Comment:
This is not going to work yet, because the SLSA verifier is not aware of the `https://github.com/slsa-framework/slsa-github-generator/.github/workflows/provenance.yml` yet.

We need to decide how the verification work, because the SLSA level is different. For example, we may need https://github.com/slsa-framework/slsa-verifier/issues/21 to be resolved. (Not clear yet how to express that certain fields only are forgeable, and not others, though)

Reply:
Yeah, maybe I can leave the verification bit off for now.

Reply:
Removed for now. We can discuss later how to add verifier support.

Reply:
SGTM